server/server.ts

```diff
@@ -315,6 +323,83 @@ app.get('/api/fetchlikes/:userId', authMiddleware, async (req, res, next) => {
   }
 });
 
+app.post('/api/sendemail', async (req, res, next) => {
+  const reqBody = req.body;
+  const objectData = reqBody.objectData as EmailSentForm;
+  const toAddress = objectData.toAddress;
+  const fromAddress = objectData.fromAddress;
+  const emailTitle = objectData.title;
+  const message = objectData.message;
+
+  if (!objectData) {
+    throw new ClientError(400, 'email form object can not be undefined.');
+  }
+  const recipe: Recipe = reqBody.recipe;
+  if (!recipe) {
+    res.json('recipe undefined');
```

Comment:
If you make the changes suggested above in the client, you could handle this the same way you handled `objectData`. That is, `throw new ClientError(400, 'recipe can not be undefined.');`.